rust/src/ldap/detect.rs

```diff
@@ -0,0 +1,244 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use super::ldap::{LdapTransaction, ALPROTO_LDAP};
+use crate::detect::uint::{
+    detect_parse_uint_enum, rs_detect_u8_free, rs_detect_u8_match, DetectUintData,
+};
+use crate::detect::{
+    DetectHelperBufferRegister, DetectHelperKeywordRegister, DetectSignatureSetAppProto,
+    SCSigTableElmt, SigMatchAppendSMToList,
+};
+use crate::ldap::types::{LdapMessage, ProtocolOpCode};
+
+use std::ffi::CStr;
+use std::os::raw::{c_int, c_void};
+use std::str::FromStr;
+
+pub const DETECT_LDAP_RESP_ANY: i8 = -1;
+pub const DETECT_LDAP_RESP_ALL: i8 = -2;
+
+#[derive(Debug, PartialEq)]
+pub struct DetectLdapRespData {
+    pub response: DetectUintData<u8>, //ldap response code
+    pub index: i8,
```

Comment:
May we should support indexes bigger than 255==u8::MAX

Let's try i32

Reply:
And if we can do pure rust here, you can have the struct not public, and having the index have a type that is an enum that is either `Any`, `All`, or `Index(i32)`